lib/index.test.ts

```diff
@@ -65,13 +65,14 @@ describe('validateEnvSchema', (): void => {
         validateEnvSchema(schema, container);
         expect(true).toBe(false); // should throw
       } catch (e) {
+        const err = e as EnvSchemaValidationError<typeof schema, undefined>;
```

Comment:
why these error casts? are we sure it will always throw these kinds of errors?

Reply:
You could instead define a function to check it, since `expect` isn't enought for typescript to infer the correct type.
```typescript
const assertIsEnvSchemaValidationError = <TSchema, TCustomizations = undefined>(e: unknown) assert e is EnvSchemaValidationError<TSchema, TCustomizations> => {
if(!(e instanceof EnvSchemaValidationError)){
  //throw;
}
}

...
expect(assertIsEnvSchemaValidationError(e)).not.to.throw();
```

Reply:
There is a way to narrow down using something like `e is EnvSchemaValidationError<TSchema, TCustomizations>`, I know I did some code like that without assert and it worked, but it should return a boolean I think